### ISSUE TITLE
Client preset cannot be somehow used with typescript-react-apollo plugin

### ISSUE THREAD

**Issue by simPod** [dotansimha/graphql-code-generator#10212]
### Describe the bug

I'm trying to adopt `client` preset. I'm using `typescript-react-apollo` and I'm not sure how to get apollo hooks with client preset. It generates duplicate Documents.

_This PR https://github.com/dotansimha/graphql-code-generator/pull/8500 seems that it tried to prevent using plugins with client preset but CLI gives no errors._

### Your Example Website or App

n/a

### Steps to Reproduce the Bug or Issue

Use client preset with typescript-react-apollo plugin.

### Expected behavior

No duplicated Documents generated.

### Platform

OS: macos

NodeJS: 22

graphql version: 16.3

"@graphql-codegen/add": "^5.0.3",
"@graphql-codegen/cli": "^5.0.3",
"@graphql-codegen/client-preset": "^4.5.1",
"@graphql-codegen/fragment-matcher": "^5.0.2",
"@graphql-codegen/typescript": "^4.1.2",
"@graphql-codegen/typescript-document-nodes": "^4.0.12",
"@graphql-codegen/typescript-operations": "^4.4.0",
"@graphql-codegen/typescript-react-apollo": "^4.3.2",

**Comment by simPod** [comment]
Ah ok, so I should migrate to native apollo hooks first. Thanks

**Comment by eddeee888** [comment]
That's right 🙂 I'll close this issue since there's a recommended path forward. Please reach out anytime!
